src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/RazorFormatOnTypeProvider.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting
+{
+    internal abstract class RazorFormatOnTypeProvider
+    {
+        public abstract string TriggerCharacter { get; }
```

Comment:
Will we ever want to have multiple providers for the same trigger character? In other words, will we ever want to do two separate set of edits for the same trigger character?

Reply:
Yes. For instance, the newlines case requires separate providers for HTML, Text Tags, Razor curly braces, C# curly braces.